pkg/scheduler/reconciliation/reconciliation.go

```diff
@@ -22,6 +23,7 @@ type Repository interface {
 	RemoveReconciliationByRuntimeID(runtimeID string) error
 	RemoveReconciliationBySchedulingID(schedulingID string) error
 	RemoveReconciliationsBySchedulingID(schedulingIDs []string) error
+	RemoveReconciliations(filter Filter) error
```

Comment:
```suggestion
	RemoveReconciliationsByFilter(filter Filter) error
```
Not a must-have but would align with the rest